src/main/java/org/dasein/cloud/ResourceNotFoundException.java

```diff
@@ -0,0 +1,39 @@
+package org.dasein.cloud;
+
+import javax.annotation.Nonnegative;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * User: daniellemayne
+ * Date: 15/12/2015
+ * Time: 16:20
+ */
+public class ResourceNotFoundException extends CloudException {
+    public ResourceNotFoundException(@Nonnull String msg) {
```

Comment:
I think this exception only needs a single constructor taking two parameters: String resourceType, String resourceId. We should also review other exceptions and try and leave only the meaningful constructors there.